### PR TITLE
Bump libcontainer to 5765dcd086eb0584c0e2eaff9a3ac97b467a98e6

### LINKED ISSUE #6880: Permission denied on /dev/stderr
The following Docker run command returns the showed error message: 

    docker run -u $(id -u) -v $PWD:$PWD -w $PWD ubuntu sh -c 'echo Hola > /dev/stderr'  
    sh: 1: cannot create /dev/stderr: Permission denied

Removing the `-u` command option it works fine. 

This issue make impossible to run that apps that use the `/dev/stderr` output stream to report warning messages or other info.

### PR DESCRIPTION
**Pull request by crosbymichael**
Fixes #6880

This bumps libcontainer to fix some STDIO permission issues and update docker with the latest changes to libcontainer and it's configuration.

### CODE REVIEW

**Comment by crosbymichael** [comment]
@rhvgoyal 

This brings in the mount propagation changes into docker but we are having some issues when running containers with --read-only.  I think it has something due to the changes, maybe something around the root not being PRIVATE anymore or something else with the mounts and propagation flags.  Could u please take a look?

**Comment by rhvgoyal** [comment]
@crosbymichael I looked briefly at my patch and can't think how it is leading to failure. Tomorrow I will spend more time on this. First I need to understand the test which is failing and that will give more clues.

BTW, my patch does not change the propagation property of rootfs and it still continues to be PRIVATE.

**Comment by rhvgoyal** [comment]
For me --read-only works with -ti option but not without it.

"docker run --red-only fedora ls" is failing while "docker run -ti --read-only fedora ls" succeeds.

I did strace on docker daemon and somebody is doing "fchown()" on a read only file system hence -EROFS is returned.

30347 fchown(0, 0, 0)                   = -1 EROFS (Read-only file system)

I am not sure who is doing fchown() yet.

**Comment by rhvgoyal** [comment]
@mrunalp Would you have any idea.

**Comment by rhvgoyal** [comment]
Ok, setupUser() if failing in libcontainer (init_linux.go)

        if err := setupUser(config); err != nil {
                return err
        }

**Comment by mrunalp** [comment]
Fchown is probably from the user ns related two patches.

**Comment by rhvgoyal** [comment]
Following commit introduce Fchown()
commit 0dad64f7ad4d27a02aa0c6605304832e87a79974
Author: Michael Crosby <crosbymichael@gmail.com>
Date:   Fri Sep 18 13:55:49 2015 -0700

    Fix STDIO permissions when container user not root
    
    Fix the permissions of the container's main processes STDIO when the
    process is not run as the root user.  This changes the permissions right
    before switching to the specified user so that it's STDIO matches it's
    UID and GID.
    
    Add a test for checking that the STDIO of the process is owned by the
    specified user.
    
    Signed-off-by: Michael Crosby <crosbymichael@gmail.com>

**Comment by crosbymichael** [comment]
ok, thanks for taking a look at this.  I was not sure about changing the propagation of the rootfs from PRIVATE to something else or not.

**Comment by rhvgoyal** [comment]
@crosbymichael rootfs propagation mode change patches have not been merged yet. I am reworking those patches now.

**Comment by crosbymichael** [comment]
Ok, i figured out the issue, it's with `-i` and the readonly containers when there is no stdin.

```bash
michael|~ > docker run ubuntu ls -l /proc/1/fd/
total 0
lr-x------ 1 root root 64 Sep 22 17:41 0 -> /dev/null
l-wx------ 1 root root 64 Sep 22 17:41 1 -> pipe:[143849]
l-wx------ 1 root root 64 Sep 22 17:41 2 -> pipe:[143850]
lr-x------ 1 root root 64 Sep 22 17:41 3 -> /proc/1/fd
michael|~ > docker run -i ubuntu ls -l /proc/1/fd/
total 0
lr-x------ 1 root root 64 Sep 22 17:41 0 -> pipe:[142163]
l-wx------ 1 root root 64 Sep 22 17:41 1 -> pipe:[142167]
l-wx------ 1 root root 64 Sep 22 17:41 2 -> pipe:[142168]
lr-x------ 1 root root 64 Sep 22 17:41 3 -> /proc/1/fd
```

**Comment by rhvgoyal** [comment]
Oh, so without "-i", stdin is pointing to /dev/null and we try to change ownership of /dev/null instead? And that will fail as / is read only.

**Comment by crosbymichael** [comment]
Naw, I just have to either ignore it if it's `/dev/null` or find a robust way to handle this case without being too specific.

**Comment by estesp** [comment]
I assume if we are going to pick up the update to handle `/dev/null` properly (opencontainers/runc#289), we should grab whatever gets merged to handle user ns join; currently I think we are leaning towards opencontainers/runc#288?

I think the combination of those (plus the ones included already in this PR) fix all known issues with user ns from a lower layer (libcontainer) perspective.

**Comment by crosbymichael** [comment]
@icecrime @estesp @tiborvass this one should be good to go execpt for waiting on the apparmor test fixes.  I can rebase once @icecrime is done doing what he is doing.

**Comment by estesp** [comment]
Looks like switching to the vendored netlink library kills Windows daemon build.  I assume putting a dummy .go file in `netlink/nl` would fix the compilation problem, but is the `getDefaultRouteMtu` actually working today for Windows daemon operation? @jhowardmsft

**Comment by crosbymichael** [comment]
@estesp i'm fixing that in runc now

**Comment by jhowardmsft** [comment]
@estesp - No, it's not used on Windows (yet)

**Comment by tiborvass** [comment]
LGTM

**Comment by jfrazelle** [comment]
LGTM

File: hack/vendor.sh [hunk 1]
```diff
@@ -42,7 +42,7 @@ clone git github.com/endophage/gotuf 9bcdad0308e34a49f38448b8ad436ad8860825ce
 clone git github.com/jfrazelle/go 6e461eb70cb4187b41a84e9a567d7137bdbe0f16
 clone git github.com/agl/ed25519 d2b94fd789ea21d12fac1a4443dd3a3f79cda72c
 
-clone git github.com/opencontainers/runc v0.0.4 # libcontainer
+clone git github.com/opencontainers/runc aac9179bbadbf958054ce97ab368ac178140e5da # libcontainer
```
Comment (tianon) [inline hack/vendor.sh]: :cry:

**Pull request merged**